### ISSUE TITLE
Goals Bart for February

### ISSUE THREAD

**Issue by louwers** [maplibre/maplibre#218]
Aside from the defined [responsibilities](https://github.com/maplibre/maplibre/wiki/Maintainer-Native-Role) as maintainer I created some tangible goals for February.

- Become expert of the CMake build setup
  - [ ] Make an high-level overview of the current situation	
  - [x] Make a suggestion towards meaningfully improving composability (decouple platform-related CMake code from Core CMake code)
  - [ ] Implement a first step of that improvement
  - [x] Make a suggestion for handling C++ dependencies, either continue with submodules or use a package manager
- [x] New C++ Documentation with Doxygen live
- [x] Intergrate DocC (e.g. allow building DocC docs) for iOS docs, document what changes need to be made
  - [ ] Write out a bounty for changes needed to iOS codebase
- [x] Render a triangle with Metal (i.e. become somewhat familiar with Metal)
- MapLibre GL JS update
  - [x] Fix all C++ issues unrelated to rendering myself
  - [ ] Write out a bounty (or multiple) for issues related to rendering

A focus for February will be to stimulate contributions to MapLibre GL Native and to kick off the bounty program.

**Comment by louwers** [comment]
For the build setup I think I will limit myself to iOS.

**Comment by louwers** [comment]
I made some progress towards updating the MapLibre GL JS submodule. I modified the template that was generating the Objective-C tests because some were disabled by hand. That resolved the issues with these tests, but the render tests are failing. I need to spend some time next month to better understand the render tests.

@ntadej made a code reorganization proposal. I had a discussion about it with him. We want to decouple the core from the platforms. The approach I suggested was that we make it possible that the core is independently released and depend on that released version for one of the platforms. We can prototype that approach with the Qt platform before moving onto other platforms. We will continue this discussion during next week's TSC meeting. I am definitely not an expert of the build setup yet, but I made some small improvements such a removing unused or deduplicated scripts and some generated files.

I'll think about and make a new ticket for my goals of March.
